src/coreclr/jit/likelyclass.cpp

```diff
@@ -244,12 +244,27 @@ static unsigned getLikelyClassesOrMethods(LikelyClassMethodRecord*
 
                     const UINT32 numberOfClasses = min(knownHandles, maxLikelyClasses);
 
+                    double roundingError = 0.0;
                     for (size_t hIdx = 0; hIdx < numberOfClasses; hIdx++)
                     {
-                        LikelyClassMethodHistogramEntry const hc = sortedEntries[hIdx];
-                        pLikelyEntries[hIdx].handle              = hc.m_handle;
-                        pLikelyEntries[hIdx].likelihood          = hc.m_count * 100 / h.m_totalCount;
+                        LikelyClassMethodHistogramEntry const hc         = sortedEntries[hIdx];
+                        const double                          likelihood = hc.m_count * 100.0 / h.m_totalCount;
+                        pLikelyEntries[hIdx].handle                      = hc.m_handle;
+                        pLikelyEntries[hIdx].likelihood                  = (UINT32)likelihood;
+
+                        // Accumulate the rounding error
+                        roundingError += likelihood - (double)(UINT32)likelihood;
                     }
+
+                    // Distribute the rounding error, just apply it to the first entry
+                    // so we can avoid marking fallback case as e.g. 1% likely while in fact it's 0.
+                    if (roundingError >= 1.0)
+                    {
+                        assert(numberOfClasses > 0);
+                        pLikelyEntries[0].likelihood += (UINT32)roundingError;
+                        assert(pLikelyEntries[0].likelihood <= 100);
+                    }
+
```

Comment:
Would suggest just doing all integer math instead, something like

```c++
UINT32 totalLikelihood = 0;
for (...)
 {
     pLikelyEntries[hIdx].likelihood = hc.m_count * 100 / h.m_totalCount;
     totalLikelihood += pLikelyEntries[hIdx].likelihood;
}

pLikelyEntries[0].likelihood += 100 - totalLikelihood;

Reply:
Addressed. The only thing that I had to check that we don't have unknown handles, if we have them - we just assume that the missing likelihood goes to them.